apps/desktop/desktop_native/core/src/ssh_agent/mod.rs

```diff
@@ -36,19 +38,43 @@ pub struct SshAgentUIRequest {
     pub cipher_id: Option<String>,
     pub process_name: String,
     pub is_list: bool,
+    pub is_sig: bool,
+    pub is_git: bool,
+    pub is_forwarding: bool,
 }
 
 impl ssh_agent::Agent<peerinfo::models::PeerInfo> for BitwardenDesktopAgent {
-    async fn confirm(&self, ssh_key: Key, info: &peerinfo::models::PeerInfo) -> bool {
+    async fn confirm(&self, ssh_key: Key, data: &[u8], info: &peerinfo::models::PeerInfo) -> bool {
         if !self.is_running() {
             println!("[BitwardenDesktopAgent] Agent is not running, but tried to call confirm");
             return false;
         }
 
         let request_id = self.get_request_id().await;
+        let request_data = request_parser::parse_request(data);
+        if let Err(e) = request_data {
+            println!("[SSH Agent] Error while parsing request: {}", e);
+            return false;
+        }
+        let request_data = request_data.unwrap();
```

Comment:
We should avoid unwraps if we're already matching the err pattern:
```suggestion
        let request_data = match request_parser::parse_request(data) {
            Ok(data) => data,
            Err(e) => {
                println!("[SSH Agent] Error while parsing request: {}", e);
                return false;
            }
        };
```